python/pyspark/ml/linalg/__init__.py

```diff
@@ -72,7 +72,9 @@ def _convert_to_vector(l):
         return DenseVector(l)
     elif _have_scipy and scipy.sparse.issparse(l):
         assert l.shape[1] == 1, "Expected column vector"
-        csc = l.tocsc()
+        # Make sure the converted csc_matrix has sorted indices.
+        csc = l.tocsr().tocsc()
```

Comment:
I'd just call tocsc.  You can then check the attribute has_sorted_indices and call sort_indices() if needed.